vestido/vestido-storefront/modules/products/product-view.tsx

```diff
@@ -35,8 +42,15 @@ const ProductView: React.FC<ProductViewProps> = ({ itemId }) => {
   const { isAuthenticated, routeToLogin } = useAuth();
   const router = useRouter();
   const { toast } = useToast();
+  const sizeCharts = useVestidoSizeChart();
+  const sizeChartTitles = sizeCharts ? Object.keys(sizeCharts) : [];
 
   const { data: { data: item } = { data: null } } = useItem(itemId);
+
+  const sizeChartTitle = sizeChartTitles.find((x) => x === item?.sizeChart);
+
+  const sizeChartValue =
+    sizeCharts && sizeChartTitle ? sizeCharts[sizeChartTitle] : undefined;
```

Comment:
Please try to improve this.
- sizeChartIds instead of sizeChartTitles
- Make use of `useMemo()`
- Multiple intermediate variables. I feel like they can be combined to one statement inside the useMemo